### PR TITLE
docs: more release instructions

### PR DESCRIPTION
**Pull request by mistercrunch**
@dpgaspar @villebro  ^^^

### CODE REVIEW

**Comment by mistercrunch** [comment]
Oh I just saw https://github.com/apache/incubator-superset/pull/8381 ! Let's merge that one first and see if there's anything of value left in here after...

**Comment by villebro** [comment]
@mistercrunch these partly overlap stuff in #8381, can we finalize that and then rebase this?

**Comment by mistercrunch** [comment]
Agreed, yours first and then I'll merge the section about emails on top

File: RELEASING/README.md [hunk 1]
```diff
@@ -19,17 +19,21 @@ under the License.
 
 # Apache Releases
 
-You'll probably want to run these commands manually and understand what
+Until we things settle and we create scripts that streamline this,
```
Comment (dpgaspar) [inline RELEASING/README.md]: Until things settle ...

File: RELEASING/README.md [hunk 2]
```diff
@@ -19,17 +19,21 @@ under the License.
...
+synchronous communications, we recommend using the `#apache-releases` channel
 on the Superset Slack. People crafting releases and those interested in
 partaking in the process should join the channel.
 
-## Release setup
 
-First you need to setup a few things. This is a one-off and doesn't
+## The Release Process
+
+### Environment preparation for a relase
```
Comment (dpgaspar) [inline RELEASING/README.md]: ### Environment preparation for a release

**Review by kristw — APPROVED** [review]

**Review by villebro — APPROVED** [review]

**Review by dpgaspar — APPROVED** [review]

File: RELEASING/README.md [hunk 3]
```diff
@@ -134,7 +135,29 @@ Now let's ship this RC into svn's dev folder
     svn commit -m "Release ${SUPERSET_VERSION_RC}"
 ```
 
-Now you're ready to start the VOTE thread.
+### Voting
+Now you're ready to start the [VOTE] thread. Here's an example of a
+previous release vote thread:
+https://lists.apache.org/thread.html/e60f080ebdda26896214f7d3d5be1ccadfab95d48fbe813252762879@<dev.superset.apache.org>
+
+Once 3+ binding votes (by PMC members) have been cast and at
```
Comment (john-bodley) [inline RELEASING/README.md]: @mistercrunch is there a list of PMC members? If so could this be a hyperlink?

**Pull request merged**